pkg/controller/kfservice/kfservice_controller.go

```diff
@@ -159,6 +159,15 @@ func (r *ReconcileService) updateStatus(before *kfservingv1alpha1.KFService, ksv
 	} else {
 		after.Status.Canary.Name = ksvc.Status.LatestCreatedRevisionName
 	}
+	for _, traffic := range ksvc.Status.Traffic {
```

Comment:
Is it possible that default + canary wouldn't sum to 100?

Reply:
it could be 0 when default is not in ready status(does not pass readiness probe), but as soon as default is ready the traffic should sum to 100, for example when default is ready and user adds a canary which does not pass the readiness probe, 100% traffic stays with default, as soon as canary goes into ready state the traffic splits between the two and sum to 100.